src/engraving/libmscore/measure.cpp

```diff
@@ -3239,12 +3240,27 @@ void Measure::layoutMeasureElements()
                 for (s1 = s.prevActive(); s1 && s1->allElementsInvisible(); s1 = s1->prevActive()) {
                 }
                 Segment* s2;
-                for (s2 = s.nextActive(); s2; s2 = s2->nextActive()) {
-                    if (!s2->isChordRestType() && s2->element(staffIdx * VOICES)) {
-                        break;
+                if (!modernMMRest) {
+                    // Ignore any stuff before the end bar line (such as courtesy clefs)
+                    s2 = findSegment(SegmentType::EndBarLine, endTick());
+                } else {
+                    for (s2 = s.nextActive(); s2; s2 = s2->nextActive()) {
+                        if (!s2->isChordRestType() && s2->element(staffIdx * VOICES)) {
+                            break;
+                        }
                     }
                 }
-                double x1 = s1 ? s1->x() + s1->minRight() : 0;
+                double x1;
+                if (!modernMMRest && s1) {
+                    Segment* thisSeg = toRest(e)->segment();
+                    // Center with respect to the "virtual" position of where the preceding barline would be
+                    x1 = thisSeg->x() - thisSeg->minLeft() - score()->styleMM(Sid::barNoteDistance);
+                } else if (s1) {
+                    x1 = s1->x() + s1->minRight();
+                } else {
+                    x1 = 0;
+                }
```

Comment:
Here too I think we can make some simplifications:
- initialize x1 immediately with 0.0 (uninitialized variables are always scary)
- structure the `if` statements like this: 
    ```c++
    double x1 = 0.0;
    if (s1) {
         if (modernMMRest) {
             …
         } else {
             …
         }
    }
    ```

Reply:
Never mind, we've actually scrapped this part